pandas/core/generic.py

```diff
@@ -3770,10 +3773,12 @@ def as_matrix(self, columns=None):
         --------
         pandas.DataFrame.values
         """
+        warnings.warn("method ``as_matrix`` will be removed in a future version. "
+                      "Use ``values`` instead.", FutureWarning, stacklevel=2)
         self._consolidate_inplace()
         if self._AXIS_REVERSED:
```

Comment:
ok so this should be the same as below (e.g. passing ``transpose=``). make sure we still have a test on this (e.g. that passes columns).

Reply:
Ok, done.

Reply:
Ok, and ``test_as_matrix_deprecated`` has been modified to a take ``columns`` param.